### PR TITLE
should run location sync with specific vagrant sub commands only to i…

### PR DESCRIPTION
**Pull request by phuonglm**
connect #531

### CODE REVIEW

**Review by datphan — DISMISSED** [review]

File: lib/teracy-dev/extension/manager.rb [hunk 1]
```diff
@@ -52,13 +53,22 @@ def sync(extension)
           "path" => path
         })
         sync_existing = extension['path']['lookup'] == DEFAULT_EXTENSION_LOOKUP_PATH
-        Location::Manager.sync(extension['location'], sync_existing)
+
+        if Util.cmdRequireSync?
```
Comment (hoatle) [inline lib/teracy-dev/extension/manager.rb]: move this condition into Location::Manager.sync instead, add optional argument force=false; if force = true and Util.cmdRequireSync? is false => continue syncing

`
Location::Manager.sync(location, sync_existing, force=False)
`

File: lib/teracy-dev/extension/manager.rb [hunk 2]
```diff
@@ -52,13 +53,22 @@ def sync(extension)
...
       end
 
       def validate(extension)
         return unless Util.true?(extension['enabled'])
 
-        manifest = Manager.manifest(extension)
+        begin
+          manifest = Manager.manifest(extension)
+        rescue Errno::ENOENT
+          @logger.error("manifest for extension #{extension['path']['extension']} not found, seem you not sync all required extension. Please run vagrant <status|up|reload> to sync it first.")
```
Comment (hoatle) [inline lib/teracy-dev/extension/manager.rb]: @logger.error("manifest for extension #{extension['path']['extension']} not found, seem that you haven't synced all the required extensions. Please run vagrant <status|up|reload> to sync them first")

File: lib/teracy-dev/loader.rb [hunk 3]
```diff
@@ -125,7 +125,7 @@ def sync_teracy_dev_entry(location)
         return
       end
 
-      if Location::Manager.sync(location, location['sync']) == true
+      if Util.cmdRequireSync? && Location::Manager.sync(location, location['sync']) == true
```
Comment (hoatle) [inline lib/teracy-dev/loader.rb]: don't change this, Location::Manager.sync should be updated only

File: lib/teracy-dev/util.rb [hunk 4]
```diff
@@ -19,6 +19,10 @@ def self.exist?(value)
       exist
     end
 
+    def self.cmdRequireSync?
```
Comment (hoatle) [inline lib/teracy-dev/util.rb]: move this into Location::Manager.sync as private for now

**Review by hoatle — CHANGES_REQUESTED** [review]
@phuonglm pls check my comments

**Comment by phuonglm** [comment]
I updated the PR.

File: lib/teracy-dev/location/manager.rb [hunk 5]
```diff
@@ -21,6 +24,10 @@ def self.sync(location, sync_existing = true)
         @@logger.debug("sync finished in #{timer_end - timer_start}s with updated: #{updated}, location: #{location}")
         updated
       end
+
+      def self.cmdRequireSync?
```
Comment (hoatle) [inline lib/teracy-dev/location/manager.rb:28]: follow convention for method name (snake case instead of camel case), `sync_required?` is good name to me

**Review by hoatle — APPROVED** [review]
LGTM, I'll update `cmdRequireSync` to `sync_required?`

**Pull request merged**